src/main.tf

```
@@ -0,0 +1,33 @@
+locals {
+  geo_location_map = {
+    "Africa"        = "southafricanorth"
+    "Asia Pacific"  = "southeastasia"
+    "Australia"     = "australiaeast"
+    "Brazil"        = "brazilsouth"
+    "Canada"        = "canadacentral"
+    "Europe"        = "westeurope"
+    "France"        = "francecentral"
+    "Germany"       = "germanywestcentral"
+    "India"         = "centralindia"
+    "Japan"         = "japaneast"
+    "Korea"         = "koreacentral"
+    "Norway"        = "norwayeast"
+    "Switzerland"   = "switzerlandnorth"
+    "UAE"           = "uaenorth"
+    "UK"            = "uksouth"
+    "United States" = "centralus"
+  }
+  location = lookup(local.geo_location_map, var.location, "")
```

A user can add something to the `massdriver.yaml` and forget to add it to the list above. In that case validation will pass but `location` will be `""` and provisioning will fail.

Before I really learned JSON schema, I think I did this in a few places in GCP as well. Mostly for enums in static lists like availability `Highly Available => HIGHLY_AVAILABLE`. I think for Firestore, I do this with locations and if that yaml changes it would be the same issue. I'll plop a ticket in the backlog for me to check / update Firestore.

---

Would the best approach be to remove the `""` from L20?

---

It sounds like these all map to a specific thing, so there shouldn't be a missing one. Only in the case that a new data location is added and we don't pick a region for the resource group.

💡 instead of `""` we could have that be a default location, but that might be a future landmine. Like, can I have a resource group in Asia but a Comm Service in the `UK`?

---

I will test this potential outcome and let you know

---

Yes, it is possible for the Azure Communication Service to deploy to a geography like UK and for the resource group it's attached to be located in somewhere like Japan East.

---

if its blank because someone entered something weird, we should let it crash... they should get a validation error from the API, so highly unlikely

---

Ok, I'll leave as is then.